core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,910 @@
+            .build();
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery(spark.dialect(), spark.sql())
+            .create();
+
+    assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+
+    ViewVersion viewVersion = view.currentVersion();
+    assertThat(viewVersion.representations()).hasSize(1).containsExactly(spark);
+
+    SQLViewRepresentation updatedSpark =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from ns.updated_tbl")
+            .dialect("spark")
+            .build();
+
+    // only update the SQL for spark
+    view.replaceVersion()
+        .withSchema(SCHEMA)
+        .withDefaultNamespace(identifier.namespace())
+        .withQuery(updatedSpark.dialect(), updatedSpark.sql())
+        .commit();
+
+    // history and view versions should reflect the changes
+    View updatedView = catalog().loadView(identifier);
+    assertThat(updatedView.history())
+        .hasSize(2)
+        .element(0)
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(viewVersion.versionId());
+    assertThat(updatedView.history())
+        .element(1)
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(updatedView.currentVersion().versionId());
+    assertThat(updatedView.versions())
+        .hasSize(2)
+        .containsExactly(viewVersion, updatedView.currentVersion());
+
+    // updated view should have the new SQL
+    assertThat(updatedView.currentVersion().representations())
+        .hasSize(1)
+        .containsExactly(updatedSpark);
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void replaceViewVersionErrorCases() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    SQLViewRepresentation trino =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from ns.tbl")
+            .dialect("trino")
+            .build();
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery(trino.dialect(), trino.sql())
+            .create();
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isTrue();
+
+    // empty commits are not allowed
+    assertThatThrownBy(() -> view.replaceVersion().commit())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Cannot replace view without specifying a query");
+
+    // schema is required
+    assertThatThrownBy(
```

Comment:
There should be an assertion for default namespace as well, right?

Reply:
yep, added one that makes sure the default namespace is set